core/server/policy_violations.go

```diff
@@ -0,0 +1,267 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/hashicorp/go-multierror"
+	"github.com/weaveworks/weave-gitops/core/clustersmngr"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/core"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	v1 "k8s.io/api/core/v1"
+	k8sFields "k8s.io/apimachinery/pkg/fields"
+	k8sLabels "k8s.io/apimachinery/pkg/labels"
+	sigsClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type validationList struct {
+	Validations []*pb.PolicyValidation
+	Token       string
+	Errors      []*pb.ListError
+}
+
+func (cs *coreServer) ListPolicyValidations(ctx context.Context, m *pb.ListPolicyValidationsRequest) (*pb.ListPolicyValidationsResponse, error) {
+	var respErrors []*pb.ListError
+	clustersClient, err := cs.clustersManager.GetImpersonatedClient(ctx, auth.Principal(ctx))
```

Comment:
in the `ListObjects` they check on cluster name and accordingly use either `GetImpersonatedClientForCluster` or `GetImpersonatedClient`...do you know why it's being done like that there and is there a reason why you're not following the same approach?

Reply:
will follow the same approch